pkg/extensions/tbtc/recovery/recovery_test.go

```diff
@@ -0,0 +1,147 @@
+	if expectedSerializedMessageTransaction != serializedMessageTransaction {
+		t.Errorf(
+			"expectedSerializedMessageTransaction did not match the expectation\nexpected: %s\nactual:   %s",
+			expectedSerializedMessageTransaction,
+			serializedMessageTransaction,
+		)
+	}
+}
+
+func Test_BuildSignedTransactionHexString(t *testing.T) {
+	unsignedTransaction := wire.NewMsgTx(wire.TxVersion)
+	previousOutputTransactionHash, err := chainhash.NewHashFromStr(
+		"0b99dea9655f219991001e9296cfe2103dd918a21ef477a14121d1a0ba9491f1",
+	)
+	if err != nil {
+		t.Errorf("Something went wrong generating the previousOutputTransactionHash: %v", err)
+	}
+	unsignedTransaction.AddTxIn(wire.NewTxIn(
+		wire.NewOutPoint(previousOutputTransactionHash, 0),
+		nil,
+		nil,
+	))
+
+	for _, txValue := range []int{100, 100, 100} {
+		unsignedTransaction.AddTxOut(wire.NewTxOut(
+			int64(txValue),
+			nil,
+		))
+	}
+	curve := elliptic.P224()
+	privateKey := cecdsa.PrivateKey{
+		PublicKey: cecdsa.PublicKey{
+			Curve: curve,
+			X:     bigIntFromString(t, "828612351041249926199933036276541218289243364325366441967565889653"),
+			Y:     bigIntFromString(t, "985040320797760939221216987624001720525496952574017416820319442840"),
+		},
+		D: bigIntFromString(t, "13944630350777130481687329643969358734978178969757056469087366016470"),
+	}
+	signedTransactionHex, err := BuildSignedTransactionHexString(
+		unsignedTransaction,
+		&ecdsa.Signature{
+			R:          big.NewInt(int64(3)),
+			S:          big.NewInt(int64(7)),
+			RecoveryID: 1,
+		},
+		&privateKey.PublicKey,
+	)
+	if err != nil {
+		t.Errorf("Something went wrong building the signed transaction string: %v", err)
+	}
+
+	signedTransactionBytes, _ := hex.DecodeString(signedTransactionHex)
+	signedTransaction := wire.NewMsgTx(wire.TxVersion)
+	signedTransaction.BtcDecode(bytes.NewReader(signedTransactionBytes), 1, wire.WitnessEncoding)
+	serializedSignedTransactionBytes, _ := json.Marshal(signedTransaction)
+	serializedSignedTransaction := string(serializedSignedTransactionBytes)
+	expectedSerializedSignedTransaction := "{\"Version\":1,\"TxIn\":[{\"PreviousOutPoint\":{\"Hash\":[241,145,148,186,160,209,33,65,161,119,244,30,162,24,217,61,16,226,207,150,146,30,0,145,153,33,95,101,169,222,153,11],\"Index\":0},\"SignatureScript\":\"\",\"Witness\":[\"MAYCAQMCAQcB\",\"AgAAAAAH3j67ZA0rAhWQwJ1ec5WX0C2TkiTSJ6F0A2B1\"],\"Sequence\":4294967295}],\"TxOut\":[{\"Value\":100,\"PkScript\":\"\"},{\"Value\":100,\"PkScript\":\"\"},{\"Value\":100,\"PkScript\":\"\"}],\"LockTime\":0}"
```

Comment:
Oh my. This is… Shorter than checking the fields one by one, but I wonder how painful this error message would be to decode if something went wrong and the status failed. I dig it as a ‘make sure nothing too unexpected happened’, but I feel like we should check TxIn (as below), then check TxOut (just the value and script, but not the witness), and finally check the JSON serialization. That means if there are issues in specific places we see them separately, and then if there's something real funky going on that affects the JSON output we can handle it separately.

Reply:
Of note: I don't think that should block this PR or any further work---this test is Good Enough™ as-is. If we want to deal with it in the immediate term, we can, but we can also look at it again whenever we roll around to this again.

Reply:
Thoughts around hardcoding the unsigned input and expected signed transactions serialized as hex strings? We could then simply compare results as strings and in case of a mismatch, we can recover a struct and print a pretty diff. Example below.

```go
import "github.com/google/go-cmp/cmp"

func TestBuildSignedTransactionHexString(t *testing.T) {
	unsignedTxHex := "01000000000101f19194baa0d12141a177f41ea218d93d10e2cf96921e009199215f65a9de990b000000000000000000039003fc0100000000160014a405e97c9e2efdaed32709356655ea03fc1f2a8c9003fc0100000000160014f9974ebea1ca5d6f95fb9f5509f8b3e7bb0047269003fc010000000016001495c28deefd325d2d2fc24c5ac829376dccf520e0024a00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002100000000000000000000000000000000000000000000000000000000000000000000000000"
	expectedSignedTx := "01000000000101f19194baa0d12141a177f41ea218d93d10e2cf96921e009199215f65a9de990b000000000000000000039003fc0100000000160014a405e97c9e2efdaed32709356655ea03fc1f2a8c9003fc0100000000160014f9974ebea1ca5d6f95fb9f5509f8b3e7bb0047269003fc010000000016001495c28deefd325d2d2fc24c5ac829376dccf520e0020930060201030201070121020000000007de3ebb640d2b021590c09d5e739597d02d939224d227a17403607500000000"

	publicKey := &cecdsa.PublicKey{
		Curve: elliptic.P224(),
		X:     bigIntFromString(t, "828612351041249926199933036276541218289243364325366441967565889653"),
		Y:     bigIntFromString(t, "985040320797760939221216987624001720525496952574017416820319442840"),
	}

	signature := &ecdsa.Signature{
		R:          big.NewInt(int64(3)),
		S:          big.NewInt(int64(7)),
		RecoveryID: 1,
	}

	unsignedTx := decodeTransaction(t, unsignedTxHex)

	signedTxHex, err := BuildSignedTransactionHexString(
		unsignedTx,
		signature,
		publicKey,
	)
	if err != nil {
		t.Fatalf("failed to build signed transaction: %v", err)
	}

	if signedTxHex != expectedSignedTx {
		t.Errorf(
			"invalid signed transaction\n- actual\n+ expected\n%s",
			cmp.Diff(decodeTransaction(t, signedTxHex), decodeTransaction(t, expectedSignedTx)))
	}
}

func decodeTransaction(t *testing.T, txHex string) *wire.MsgTx {
	txBytes, err := hex.DecodeString(txHex)
	if err != nil {
		t.Fatalf("failed to decode transaction [%s]: [%v]", txHex, err)
	}
	tx := wire.NewMsgTx(0)
	tx.BtcDecode(bytes.NewReader(txBytes), wire.ProtocolVersion, wire.WitnessEncoding)

	return tx
}
```

Failure would look similiar to:
```
=== RUN   TestBuildSignedTransactionHexString
    recovery_test.go:142: invalid signed transaction
        - actual
        + expected
          &wire.MsgTx{
                Version: 1,
                TxIn: []*wire.TxIn{
                        &{
                                PreviousOutPoint: {Hash: {0xf1, 0x91, 0x94, 0xba, ...}},
                                SignatureScript:  {},
                                Witness: wire.TxWitness{
                                        {
                                                0x30,
        -                                       0x06,
        +                                       0x07,
                                                0x02,
                                                0x01,
                                                ... // 5 identical elements
                                        },
                                        {0x02, 0x00, 0x00, 0x00, ...},
                                },
                                Sequence: 0,
                        },
                },
                TxOut:    {&{Value: 33293200, PkScript: {0x00, 0x14, 0xa4, 0x05, ...}}, &{Value: 33293200, PkScript: {0x00, 0x14, 0xf9, 0x97, ...}}, &{Value: 33293200, PkScript: {0x00, 0x14, 0x95, 0xc2, ...}}},
                LockTime: 0,
          }
--- FAIL: TestBuildSignedTransactionHexString (0.00s)
FAIL
FAIL    github.com/keep-network/keep-ecdsa/pkg/extensions/tbtc/recovery 0.241s
FAIL
```